cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java

```diff
@@ -211,19 +214,46 @@ private void addCustomerSelectionClaimsWhenUserHasOnePossibleLoginOrLoggedInWith
         List<AttributeType> claims) {
 
         authenticationInfo.getCurrentCustomerId()
-            .ifPresent(customerId -> claims.addAll(customerSelectionClaims(authenticationInfo, customerId)));
+            .ifPresentOrElse(generateCustomerSelectionClaimsFromAuthentication(authenticationInfo, claims),
+                             clearCustomerSelectionClaimsWhenCustomerIsAmbiguous(claims));
+    }
+
+    private Runnable clearCustomerSelectionClaimsWhenCustomerIsAmbiguous(List<AttributeType> claims) {
+        return () -> claims.addAll(emptyCustomerSelectionClaims());
+    }
+
+    private Consumer<String> generateCustomerSelectionClaimsFromAuthentication(
+        AuthenticationInformation authenticationInfo,
+        List<AttributeType> claims) {
+        
+        return customerId -> claims.addAll(customerSelectionClaims(authenticationInfo, customerId));
+    }
+
+    private List<AttributeType> emptyCustomerSelectionClaims() {
```

Comment:
overwriteAllCustomerSelectionClaimsWithNull

Reply:
Fixed and merged!